client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/Connection.spec.ts

```diff
@@ -191,6 +191,87 @@ describe("Connection", () => {
         done();
     });
 
+    describe("negotiate request", () =>
+    {
+        let negotiateUrl: string;
+        let connection: HttpConnection;
+        let options: IHttpConnectionOptions = {
+            httpClient: <IHttpClient>{
+                post(url: string): Promise<string> {
+                    negotiateUrl = url;
+                    connection.stop();
+                    return Promise.resolve("{}");
+                },
+                get(url: string): Promise<string> {
+                    connection.stop();
+                    return Promise.resolve("");
+                }
+            },
+            logging: null
+        } as IHttpConnectionOptions;
+
+        it("preserves user query with / before query", async done =>
+        {
+            connection = new HttpConnection("http://tempuri.org/endpoint/?q=my/Data", options);
+
+            try {
+                await connection.start();
+                done();
+            }
+            catch (e) {
+                fail();
+                done();
+            }
+
+            expect(negotiateUrl).toBe("http://tempuri.org/endpoint/negotiate?q=my/Data");
+        });
+
+        it("preserves user query without / before query", async done => {
+            connection = new HttpConnection("http://tempuri.org/endpoint?q=my/Data", options);
+
+            try {
+                await connection.start();
+                done();
+            }
+            catch (e) {
+                fail();
+                done();
+            }
+
+            expect(negotiateUrl).toBe("http://tempuri.org/endpoint/negotiate?q=my/Data");
+        });
+
+        it("adds negotiate after path without /", async done => {
+            connection = new HttpConnection("http://tempuri.org/endpoint", options);
+
+            try {
+                await connection.start();
+                done();
+            }
+            catch (e) {
+                fail();
+                done();
+            }
+
+            expect(negotiateUrl).toBe("http://tempuri.org/endpoint/negotiate");
+        });
+
+        it("adds negotiate after path with /", async done => {
+            connection = new HttpConnection("http://tempuri.org/endpoint/", options);
```

Comment:
do we need a case for an url without trailing '/' ? (e.g. `http://tempuri.org/endpoint`)?

Reply:
Thats the test directly above